### PR TITLE
Split Tanstack hook files (part 2)

### PR DESCRIPTION
**Pull request by nop33**
To avoid the PR from growing too much I'll continue working on another one. This PR focuses once again in cleaning up the code, creating a more dev friendly file structure, and optimizing by moving computations inside `queryFn`s and extracting `combine` functions.

Current file structure

image

### CODE REVIEW

File: apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorth.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { TokenId } from '@/types/tokens'
+
+interface AddressesTokensWorth {
+  data: Record<TokenId, number | undefined>
+  isLoading: boolean
+}
+
+const useAddressesTokensWorth = (addressHash?: AddressHash): AddressesTokensWorth => {
+  const { data: tokenPrices, isLoading: isLoadingTokenPrices } = useAddressesTokensPrices()
+  const { data: tokensWithPrice, isLoading: isLoadingTokensWithPrice } = useAddressesPricedFTs(addressHash)
```
Comment (mvaivre) [inline apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorth.ts:38]: Just a random thought as I'm reading this, I wonder if it wouldn't be clearer and in the end more efficient (?) to get rid of the very specific `useAddressesPricedFTs` and instead have a hook that returns the the addresses tokens with their price (when available) 🤔 
It's linked to a previous discussion we had, about maybe having a simple to use `useAddressesTokens` which is a dictionary of tokens grouped by type...

Comment (mvaivre) [reply apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorth.ts:38]: + a variant hook which return a flat array with the tokens.

Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorth.ts:38]: Discussed on Slack

File: packages/shared/src/types/utils.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// Useful when we want to create type aliases for types that are just a string
+export type StringAlias = { _?: never }
```
Comment (mvaivre) [inline packages/shared/src/types/utils.ts:20]: I don't get it. Could you explain on Slack / on a call? 🙃

Comment (nop33) [reply packages/shared/src/types/utils.ts:20]: Sure!

File: apps/desktop-wallet/src/api/apiDataHooks/useAddressesUnlistedTokenTypes.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      if (!tokenIdsByTypeChunk) return tokenIdsByTypeMerged
+
+      tokenIdsByTypeMerged.fungible = tokenIdsByTypeMerged.fungible.concat(tokenIdsByTypeChunk?.fungible)
+      tokenIdsByTypeMerged['non-fungible'] = tokenIdsByTypeMerged.fungible.concat(tokenIdsByTypeChunk?.['non-fungible'])
+      tokenIdsByTypeMerged['non-standard'] = tokenIdsByTypeMerged.fungible.concat(tokenIdsByTypeChunk?.['non-standard'])
+
+      return tokenIdsByTypeMerged
+    },
+    {
+      [explorer.TokenStdInterfaceId.Fungible]: [],
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/useAddressesUnlistedTokenTypes.ts]: Agreed! Fixed!

File: apps/desktop-wallet/src/api/apiDataHooks/useAddressesUnlistedTokenTypes.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+export default useAddressesUnlistedTokenTypes
+
+const combine = (results: UseQueryResult<TokenTypesQueryFnData>[]) => ({
+  data: results.reduce(
+    (tokenIdsByTypeMerged, { data: tokenIdsByTypeChunk }) => {
+      if (!tokenIdsByTypeChunk) return tokenIdsByTypeMerged
+
+      tokenIdsByTypeMerged.fungible = tokenIdsByTypeMerged.fungible.concat(tokenIdsByTypeChunk?.fungible)
+      tokenIdsByTypeMerged['non-fungible'] = tokenIdsByTypeMerged.fungible.concat(tokenIdsByTypeChunk?.['non-fungible'])
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/useAddressesUnlistedTokenTypes.ts]: WOOPS! Fixed 😅

File: apps/desktop-wallet/src/api/apiDataHooks/useAddressesUnlistedTokenTypes.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import useAddressesListedFTs from '@/api/apiDataHooks/useAddressesListedFTs'
+import { combineIsLoading } from '@/api/apiDataHooks/utils'
+import { tokenTypesQuery, TokenTypesQueryFnData } from '@/api/queries/tokenQueries'
+
+interface AddressesUnlistedTokenTypes {
+  data: TokenTypesQueryFnData
+  isLoading: boolean
+}
+
+const useAddressesUnlistedTokenTypes = (addressHash?: AddressHash): AddressesUnlistedTokenTypes => {
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/useAddressesUnlistedTokenTypes.ts]: Fixed!

File: apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorthTotal.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import { AddressHash } from '@alephium/shared'
+import { useMemo } from 'react'
+
+import useAddressesTokensWorth from '@/api/apiDataHooks/useAddressesTokensWorth'
+
+const useAddressesTokensWorthTotal = (addressHash?: AddressHash) => {
+  const { data: tokensWorth, isLoading } = useAddressesTokensWorth(addressHash)
+
+  const totalTokensWorth = useMemo(
+    () => Object.values(tokensWorth).reduce((acc, tokenWorth) => (tokenWorth ?? 0) + (acc ?? 0), 0),
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/useAddressesTokensWorthTotal.ts:28]: Fixed!

File: apps/desktop-wallet/src/api/queries/priceQueries.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  symbol: string
+  price: number
+}[]
+
+const TOKEN_PRICES_KEY = 'tokenPrices'
+
+export const tokensPriceQuery = ({ symbols, currency }: TokensPriceQueryProps) => {
+  const getQueryOptions = (_symbols: TokensPriceQueryProps['symbols']) =>
+    queryOptions({
+      queryKey: [TOKEN_PRICES_KEY, 'currentPrice', _symbols, { currency }],
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/queries/priceQueries.ts:40]: Discussed on Slack to keep an eye in the tanstack devtools for order change

File: apps/desktop-wallet/src/api/queries/tokenQueries.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+import { TokenId } from '@/types/tokens'
+
+export type TokenTypesQueryFnData = Record<explorer.TokenStdInterfaceId, TokenId[]>
+
+const StdInterfaceIds = Object.values(explorer.TokenStdInterfaceId)
+
+export const tokenTypesQuery = (ids: TokenId[]) =>
+  queryOptions({
+    queryKey: ['tokens', 'type', ids],
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/queries/tokenQueries.ts:31]: Discussed on Slack to keep an eye in the tanstack devtools for order change

File: packages/shared/src/types/utils.ts (hunk 2 above)
Comment (nop33) [reply packages/shared/src/types/utils.ts:20]: We have a type somewhere:

```ts
type AddressHash = string
```

Now, assume there’s a prop:

```
addressHash: AddressHash
```

When you hover your mouse over it you will see screenshot 1.

But if we force TypeScript to think that this is not simply a string, then we can trick it to display the actual alias that we want (screenshot 2)

image

image

**Review by mvaivre — APPROVED** [review]
Lessgoo

**Pull request merged**